keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -0,0 +1,912 @@
+The following table is indicating what conditions are currently available (`X`) and what conditions will be added (`A`).
+
+|                                    | Waiting | Running | Progressing | Available |  ReplicaFailure | Suspended | Failed | Complete |
+| ---------------------------------  | --------| ------- | ----------- | --------- | --------------- | --------- | ------ | -------- |
+| ReplicaSet & ReplicationController | -       | -       | -           | A         | X               | -         | -      | -        |
+| Deployment                         | -       | -       | X           | X         | X               | -         | -      | -        |
+| StatefulSet                        | -       | -       | A           | A         | -               | -         | -      | -        |
+| DaemonSet                          | -       | -       | A           | A         | -               | -         | -      | -        |
+| Job                                | A       | A       | -           | -         | -               | X         | X      | X        |
+| CronJob                            | -       | -       | -           | -         | -               | -         | -      | -        |
+
+#### Progressing
+Kubernetes marks a DaemonSet or Stateful as `progressing` when:
+- The DaemonSet or StatefulSet is created
+- The DaemonSet or StatefulSet is scaling up or scaling down
+- New DaemonSet or StatefulSet pods become Ready or available
+
+#### Complete
+Kubernetes marks a DaemonSet, ReplicaSet or Stateful as `complete` when it has the following characteristics:
+
+- All of the replicas/pods associated with the DaemonSet or StatefulSet have been updated to the latest version you've specified, meaning any updates you've requested have been completed.
+- All of the replicas/pods associated with the DaemonSet, ReplicaSet or StatefulSet are available.
+- No old or mischeduled replicas/pods for the DaemonSet, ReplicaSet or Stateful are running.
+
+#### Failed
+In order to introduce this condition we need to come up with a new field called `.spec.progressDeadlineSeconds` which denotes the number of seconds the controller waits before indicating(in the workload controller status) that the controller progress has stalled.
+
+There are many factors that can cause failure to happen like:
+- Insufficient quota
+- Readiness probe failures
+- Image pull errors
+- Failed to create/delete pod
+
+See the [Kubernetes API Conventions](https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) for more information on status conditions
+
+Because of the number of changes that are involved as part of this effort, we are thinking of a phased approach where we introduce these conditions to DaemonSet controller behind a featuregate flag first in one release and then make similar changes to ReplicaSet and StatefulSet controller.
+This also needs some code refactoring of existing conditions for Deployment controller.
```

Comment:
That's fine, but in that case:
1. make the alpha longer in `kep.yaml`.
2. keep a single feature-gate, I don't see a reason why we'd need multiple.

Reply:
done